### ISSUE TITLE
JupyterLab sometimes doesn't show errors when engines throw Exception

### ISSUE THREAD

**Issue by sahil1105** [ipython/ipyparallel#707]
This might not be related to IPyParallel itself, rather IPython/JupyterLab, but I was hoping to get some feedback on how we could go about working around it.

When we have a lot of engines (and sometimes even when there's only a few), and there's exceptions on all of them, it often doesn't show them, and just say `X errors` instead (`X` is usually number of engines).

<img width="413" alt="Screen Shot 2022-04-30 at 1 28 57 PM" src="https://user-images.githubusercontent.com/14302941/167273573-434df729-dde7-499d-b452-fc80d14ec293.png">

```python
%%px
def test(arg1):
    print(arg1)
    
arg2 = "hi"
test(arg1)
```

I don't have a reliable reproducer unfortunately, since the behavior itself isn't consistent from what I've observed.

**Comment by sahil1105** [comment]
Thanks @minrk! Will try it out.

cc @hadia206

**Comment by sahil1105** [comment]
Hi @minrk, I saw this behavior recently, and running `%pxresult` indeed displayed the whole result. Is there something we could do in IPyParallel to fix it?

**Comment by minrk** [comment]
Yes, it's definitely a bug. I'm _pretty sure_ it is the assumption in `AlreadyDisplayed` that the output has been displayed by the streaming, but this has not happened (why is unclear). We need to update the logic there to either:

1. _ensure_ streaming displays the error (maybe an output timeout is being reached and giving up), or
2. _check_ if the error has been displayed, and only hide the duplicate if it has

#716 does a coarse version of this, where it should re-display errors if less than all outputs have been handled. It could result in at least partial duplicates, but duplicates are better than no feedback!

**Comment by sahil1105** [comment]
> Yes, it's definitely a bug. I'm _pretty sure_ it is the assumption in `AlreadyDisplayed` that the output has been displayed by the streaming, but this has not happened (why is unclear). We need to update the logic there to either:
> 
> 1. _ensure_ streaming displays the error (maybe an output timeout is being reached and giving up), or
> 2. _check_ if the error has been displayed, and only hide the duplicate if it has
> 
> #716 does a coarse version of this, where it should re-display errors if less than all outputs have been handled. It could result in at least partial duplicates, but duplicates are better than no feedback!

Makes sense, thanks @minrk!
